src/App.js

```diff
@@ -13,37 +13,57 @@ import { contextApi } from "./contextApi";
 function App() {
   const [products, setProducts] = useState([]);
   const [userData, setUserData] = useState(false);
+  const [loading, setLoading] = useState(false);
   useEffect(() => {
-    axios.get("http://localhost:3001/allProducts").then((res) => {
-      setProducts(res.data);
-    });
+    setLoading(true);
+    axios
+      .get("http://localhost:3001/allProducts")
+      .then((res) => {
+        setProducts(res.data);
```

Comment:
You can extract the axios request into a separate file, consider creating service that will contain all the relevant function sand export them